core/src/main/java/tachyon/worker/DataServerMessage.java

```diff
@@ -352,11 +388,36 @@ public void send(SocketChannel socketChannel) throws IOException {
   }
 
   /**
+   * copy message header and data to buffer
+   * 
+   * @param buffer
+   *          The buffer to copy message into
+   */
+  public void copyMsgToBuffer(ByteBuffer buffer) {
```

Comment:
why not expand buffer to contain all of `mHeader` and `mData`. In current implementation, if buffer size is small, mData will be truncated.